src/request/browser.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Make a request using `XMLHttpRequest`.
+ *
+ * @param   {Promise} Promise
+ * @param   {String}  method
+ * @param   {String}  url
+ * @param   {String}  body
+ * @param   {Object}  headers
+ * @returns {Promise}
+ */
+module.exports = function request (Promise, method, url, body, headers) {
+  return new Promise(function (resolve, reject) {
+    var xhr = new window.XMLHttpRequest()
+
+    xhr.open(method, url)
+
+    xhr.onload = function () {
+      return resolve({
+        status: xhr.status,
+        body: xhr.responseText
```

Comment:
That's very bad. Some servers may send important meta information through headers, and we must allow clients to read headers.